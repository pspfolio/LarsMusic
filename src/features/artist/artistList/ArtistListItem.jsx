import React from 'react';
import styled from 'styled-components';
import ArtistListItemActions from './ArtistListItemActions';
import ArtistGenres from '../artistGenres/ArtistGenres';

const ArtistCard = styled.section`
  width: 296px;
  display: flex;
  flex-direction: column;
  box-shadow: 0 7px 14px rgba(50, 50, 93, 0.1);
  border-radius: 10px;
  background-color: #fff;
  margin-bottom: 24px;
`;

const CardArtistImage = styled.div`
  width: 100%;
  height: 194px;
  background-image: url(${props => props.url});
  background-size: cover;
  border-radius: 10px 10px 0 0;
`;

const CardArtistContent = styled.div`
  display: flex;
  flex: 1;
  flex-direction: column;
  padding: 16px;
`;

const CardArtistTitle = styled.h5`
  font-weight: 400;
  font-size: 24px;
  letter-spacing: 0px;
  margin-top: 8px;
  color: rgba(0, 0, 0, 0.87);
  margin: 16px 0 0 0;
`;

const ArtistListItem = ({ images, name, genres, external_urls, id }) => {
  const image = images.find(image => image.width > 199 && image.width < 350);
  return (
    <ArtistCard>
      {image && <CardArtistImage url={image.url} />}
      <CardArtistContent>
        <CardArtistTitle>{name}</CardArtistTitle>
        <ArtistGenres genres={genres.slice(0, 3)} />
        <ArtistListItemActions externalUrl={external_urls} artistId={id} />
      </CardArtistContent>
    </ArtistCard>
  );
};

export default ArtistListItem;
