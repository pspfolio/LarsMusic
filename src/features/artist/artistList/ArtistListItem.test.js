import React from 'react';
import { shallow } from 'enzyme';
import ArtistListItem from './ArtistListItem';

describe('ArtistListItem component', () => {
  it('renders without crashing', () => {
    shallow(
      <ArtistListItem images={{ url: 'yolo' }} name="yolo" genres={['genre1', 'genre2']} external_urls={'urls'} />
    );
  });

  it('ArtistListItem renders correctly', () => {
    const wrapper = shallow(
      <ArtistListItem images={{ url: 'yolo' }} name="yolo" genres={['genre1', 'genre2']} external_urls={'urls'} />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
