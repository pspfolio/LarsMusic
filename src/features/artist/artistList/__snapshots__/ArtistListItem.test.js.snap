// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArtistListItem component ArtistListItem renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ArtistListItem
    external_urls="urls"
    genres={
        Array [
            "genre1",
            "genre2",
          ]
    }
    images={
        Object {
            "url": "yolo",
          }
    }
    name="yolo"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <styled.div
          url="yolo"
/>,
        <styled.div>
          <styled.h5>
                    yolo
          </styled.h5>
          <styled.div>
                    <styled.span>
                              genre1
                               
                              <styled.img
                                        alt="genre divider"
                                        src="round-lens-24px.svg"
                              />
                    </styled.span>
                    <styled.span>
                              genre2
                               
                              <styled.img
                                        alt="genre divider"
                                        src="round-lens-24px.svg"
                              />
                    </styled.span>
          </styled.div>
          <styled.div>
                    <styled.a
                              href="urls"
                              target="_blank"
                    >
                              Spotify
                    </styled.a>
                    <Styled(Link)
                              to="/"
                    >
                              Open
                    </Styled(Link)>
          </styled.div>
</styled.div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "url": "yolo",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <styled.h5>
              yolo
</styled.h5>,
            <styled.div>
              <styled.span>
                            genre1
                             
                            <styled.img
                                          alt="genre divider"
                                          src="round-lens-24px.svg"
                            />
              </styled.span>
              <styled.span>
                            genre2
                             
                            <styled.img
                                          alt="genre divider"
                                          src="round-lens-24px.svg"
                            />
              </styled.span>
</styled.div>,
            <styled.div>
              <styled.a
                            href="urls"
                            target="_blank"
              >
                            Spotify
              </styled.a>
              <Styled(Link)
                            to="/"
              >
                            Open
              </Styled(Link)>
</styled.div>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "yolo",
            },
            "ref": null,
            "rendered": "yolo",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <styled.span>
                  genre1
                   
                  <styled.img
                                    alt="genre divider"
                                    src="round-lens-24px.svg"
                  />
</styled.span>,
                <styled.span>
                  genre2
                   
                  <styled.img
                                    alt="genre divider"
                                    src="round-lens-24px.svg"
                  />
</styled.span>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "genre1",
                "nodeType": "class",
                "props": Object {
                  "children": Array [
                    "genre1",
                    " ",
                    <styled.img
                      alt="genre divider"
                      src="round-lens-24px.svg"
/>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "genre1",
                  " ",
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "alt": "genre divider",
                      "src": "round-lens-24px.svg",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                ],
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": "genre2",
                "nodeType": "class",
                "props": Object {
                  "children": Array [
                    "genre2",
                    " ",
                    <styled.img
                      alt="genre divider"
                      src="round-lens-24px.svg"
/>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "genre2",
                  " ",
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "alt": "genre divider",
                      "src": "round-lens-24px.svg",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                ],
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <styled.a
                  href="urls"
                  target="_blank"
>
                  Spotify
</styled.a>,
                <Styled(Link)
                  to="/"
>
                  Open
</Styled(Link)>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Spotify",
                  "href": "urls",
                  "target": "_blank",
                },
                "ref": null,
                "rendered": "Spotify",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Open",
                  "to": "/",
                },
                "ref": null,
                "rendered": "Open",
                "type": [Function],
              },
            ],
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <styled.div
            url="yolo"
/>,
          <styled.div>
            <styled.h5>
                        yolo
            </styled.h5>
            <styled.div>
                        <styled.span>
                                    genre1
                                     
                                    <styled.img
                                                alt="genre divider"
                                                src="round-lens-24px.svg"
                                    />
                        </styled.span>
                        <styled.span>
                                    genre2
                                     
                                    <styled.img
                                                alt="genre divider"
                                                src="round-lens-24px.svg"
                                    />
                        </styled.span>
            </styled.div>
            <styled.div>
                        <styled.a
                                    href="urls"
                                    target="_blank"
                        >
                                    Spotify
                        </styled.a>
                        <Styled(Link)
                                    to="/"
                        >
                                    Open
                        </Styled(Link)>
            </styled.div>
</styled.div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "url": "yolo",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <styled.h5>
                yolo
</styled.h5>,
              <styled.div>
                <styled.span>
                                genre1
                                 
                                <styled.img
                                                alt="genre divider"
                                                src="round-lens-24px.svg"
                                />
                </styled.span>
                <styled.span>
                                genre2
                                 
                                <styled.img
                                                alt="genre divider"
                                                src="round-lens-24px.svg"
                                />
                </styled.span>
</styled.div>,
              <styled.div>
                <styled.a
                                href="urls"
                                target="_blank"
                >
                                Spotify
                </styled.a>
                <Styled(Link)
                                to="/"
                >
                                Open
                </Styled(Link)>
</styled.div>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "yolo",
              },
              "ref": null,
              "rendered": "yolo",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <styled.span>
                    genre1
                     
                    <styled.img
                                        alt="genre divider"
                                        src="round-lens-24px.svg"
                    />
</styled.span>,
                  <styled.span>
                    genre2
                     
                    <styled.img
                                        alt="genre divider"
                                        src="round-lens-24px.svg"
                    />
</styled.span>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": "genre1",
                  "nodeType": "class",
                  "props": Object {
                    "children": Array [
                      "genre1",
                      " ",
                      <styled.img
                        alt="genre divider"
                        src="round-lens-24px.svg"
/>,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    "genre1",
                    " ",
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "alt": "genre divider",
                        "src": "round-lens-24px.svg",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                  ],
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": "genre2",
                  "nodeType": "class",
                  "props": Object {
                    "children": Array [
                      "genre2",
                      " ",
                      <styled.img
                        alt="genre divider"
                        src="round-lens-24px.svg"
/>,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    "genre2",
                    " ",
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "alt": "genre divider",
                        "src": "round-lens-24px.svg",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                  ],
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <styled.a
                    href="urls"
                    target="_blank"
>
                    Spotify
</styled.a>,
                  <Styled(Link)
                    to="/"
>
                    Open
</Styled(Link)>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Spotify",
                    "href": "urls",
                    "target": "_blank",
                  },
                  "ref": null,
                  "rendered": "Spotify",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Open",
                    "to": "/",
                  },
                  "ref": null,
                  "rendered": "Open",
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
